packages/discord-utilities/src/lib/regexes.ts

```diff
@@ -0,0 +1,118 @@
+import twemojiRegex from 'twemoji-parser/dist/lib/regex';
+
+/**
+ * Regex that can capture the ID in Discord Channel mentions
+ * @raw `/^(?:<#)?(\d{17,19})>?$/`
+ * @remark Capture group 1 is the ID of the channel. It is named `id`.
+ */
+export const ChannelMentionRegex = /^<#(?<id>\d{17,19})>?$/;
+
+/**
+ * Regex that matches links on the known Discord host names
+ * @raw `/(?<prefix>\w+\.)?(?<hostname>dis(?:cord)?(?:app|merch|status)?)\.(?<tld>com|g(?:d|g|ift)|(?:de(?:sign|v))|media|new|store|net)/i`
+ * @remark The regex is case insensitive
+ * @remark Capture group 1 is whatever is in the URL *before* the Discord hostname, such as `https://` and subdomains. It is named `prefix`.
+ * @remark Capture group 2 is the hostname for this URL, primarily `discord` but can also be `discordmerch`, `discordstatus`, `dis`, and `discordapp`. It is named `hostname`.
+ * @remark Capture group 3 is the Top-Level Domain *without* `.`. It is named `tld`.
+ */
+export const DiscordHostnameRegex = /(?<prefix>\w+\.)?(?<hostname>dis(?:cord)?(?:app|merch|status)?)\.(?<tld>com|g(?:d|g|ift)|(?:de(?:sign|v))|media|new|store|net)/i;
```

Comment:
```suggestion
export const DiscordHostnameRegex = /(?<subdomain>\w+\.)?(?<hostname>dis(?:cord)?(?:app|merch|status)?)\.(?<tld>com|g(?:d|g|ift)|(?:de(?:sign|v))|media|new|store|net)/i;
```

Reply:
done in 012b53881840183a57eb0785d40e7503bf617aee